src/components/spatial/Spatial.js

```diff
@@ -1,233 +1,335 @@
+ * @prop {function} getCellCoords Getter function for cell coordinates.
+ * @prop {function} getCellColor Getter function for cell color as [r, g, b] array.
+ * @prop {function} getCellPolygon
+ * @prop {function} getCellIsSelected Getter function for cell layer isSelected.
+ * @prop {function} getMoleculeColor
+ * @prop {function} getMoleculePosition
+ * @prop {function} getNeighborhoodPolygon
+ * @prop {function} updateStatus
+ * @prop {function} updateCellsSelection
+ * @prop {function} updateCellsHover
+ * @prop {function} updateViewInfo
+ * @prop {function} clearPleaseWait
+ * @prop {function} onCellClick Getter function for cell layer onClick.
  */
-export default class Spatial extends AbstractSelectableComponent {
-  constructor(props) {
-    super(props);
-    this.state.layerIsVisible = {
-      molecules: true,
-      cells: true,
-      neighborhoods: false,
-    };
-
-    // In Deck.gl, layers are considered light weight, and
-    // can be created and destroyed quickly, if the data they wrap is stable.
-    // https://deck.gl/#/documentation/developer-guide/using-layers?section=creating-layer-instances-is-cheap
-    this.moleculesData = [];
-    this.cellsData = [];
-    this.neighborhoodsData = [];
-    this.setLayerIsVisible = this.setLayerIsVisible.bind(this);
-    this.getInitialViewState = this.getInitialViewState.bind(this);
-  }
-
-  static defaultProps = {
-    clearPleaseWait: (layer) => { console.warn(`"clearPleaseWait" not provided; layer: ${layer}`); },
-    cellRadius: 10,
-  };
+export default function Spatial(props) {
+  const {
+    uuid = null,
+    view = {
+      zoom: 2,
+      target: [0, 0, 0],
+    },
+    molecules = {},
+    cells = {},
+    neighborhoods = {},
+    cellRadius = 50,
+    moleculeRadius = 10,
+    imageLayerProps = {},
+    imageLayerLoaders = {},
+    cellColors = {},
+    selectedCellIds = new Set(),
+    getCellCoords = cell => cell.xy,
+    getCellColor = cellEntry => (cellColors && cellColors[cellEntry[0]]) || DEFAULT_COLOR,
+    getCellPolygon = (cellEntry) => {
+      const cell = cellEntry[1];
+      return cell.poly.length ? cell.poly : square(cell.xy[0], cell.xy[1], cellRadius);
+    },
+    getCellIsSelected = cellEntry => (
+      selectedCellIds.size
+        ? selectedCellIds.has(cellEntry[0])
+        : true // If nothing is selected, everything is selected.
+    ),
+    getMoleculeColor = d => PALETTE[d[2] % PALETTE.length],
+    getMoleculePosition = d => [d[0], d[1], 0],
+    getNeighborhoodPolygon = (neighborhoodsEntry) => {
+      const neighborhood = neighborhoodsEntry[1];
+      return neighborhood.poly;
+    },
+    updateStatus = updateStatusWarn(COMPONENT_NAME),
+    updateCellsSelection = updateCellsSelectionWarn(COMPONENT_NAME),
+    updateCellsHover = updateCellsHoverWarn(COMPONENT_NAME),
+    updateViewInfo = updateViewInfoWarn(COMPONENT_NAME),
+    clearPleaseWait = clearPleaseWaitWarn(COMPONENT_NAME),
+    onCellClick = (info) => {
```

Comment:
Is this for allowing our collaborators to handle these themselves? Perhaps we should have a `defaultProps` object for readability.